### ISSUE TITLE
Informal Holidays in custom definition are returned even if :observed is asked

### ISSUE THREAD

**Issue by ste26054** [holidays/holidays#309]
Hello,

I am using the Holidays gem version `6.4.0` with ruby `2.3.4`

I created the following `custom_region.yaml`:

```yaml
---
months:
  1:
  - name: Informal day
    regions: [custom_region]
    mday: 1
    type: informal
```

Here, I expect `2018-01-01` to be an informal holiday.

```
2.3.4 :001 > Holidays.load_custom("custom_region.yaml")
 => {1=>[{:name=>"Informal day", :regions=>[:custom_region], :mday=>1, :type=>"informal"}]}

2.3.4 :003 > Holidays.on(Date.new(2018,1,1), :custom_region, :informal)
 => [{:date=>Mon, 01 Jan 2018, :name=>"Informal day", :regions=>[:custom_region]}] #OK

2.3.4 :004 > Holidays.on(Date.new(2018,1,1), :custom_region, :observed)
 => [{:date=>Mon, 01 Jan 2018, :name=>"Informal day", :regions=>[:custom_region]}] # ERROR: Nothing should be returned
```

So it seems Informal days in custom definitions are not working, or am I missing something?

**Comment by ste26054** [comment]
OK, I found the source of the problem. In this case, the definition should be formatted like this:

```yaml
---
months:
  1:
  - name: Informal day
    regions: [custom_region]
    mday: 1
    type: :informal
```

Note that `type: :informal` line, where `:informal` is explicitly passed as a symbol. I did not look into that first because all the existing definitions were using `type: informal`

If we do not pass a symbol, the type is passed as a string and fails the following check:
https://github.com/holidays/holidays/blob/c2463483d1289c6b5a493a5de8dab0256ff4436f/lib/holidays/finder/context/search.rb#L20

Hence all informal holidays are wrongly ignored.

I will try to make a pull request if I have time
